quinn-proto/src/config.rs

```diff
@@ -836,6 +873,45 @@ impl ServerConfig {
         self.preferred_address_v6 = address;
         self
     }
+
+    /// Maximum number of [`Incoming`][crate::Incoming] to allow to exist at a time before
+    /// rejecting new [`Incoming`][crate::Incoming] automatically.
+    ///
+    /// Should hover around zero unless connection-creating packets are being received faster than
+    /// the application is calling [`Endpoint::accept`][crate::Endpoint::accept] and dealing with
+    /// them.
```

Comment:
- A quick skim of this wording seems to suggest that it should be set to a value around 0, which is incorrect.
- Whether the number *of incoming connections* hovers around zero or not is heavily dependent on the latency with which the application layer handles them, which we can't predict.
- Incoming connections might be consumed by means other than `accept`.

How about something like:

```suggestion
    /// Maximum permitted number of [`Incoming`][crate::Incoming] connections that have not yet been [`accept`][crate::Endpoint::accept]ed or otherwise disposed
    ///
    /// While this limit is reached, new incoming connection attempts will be immediately refused. Larger values have greater worst-case memory consumption, but allow the application layer more time to handle an incoming connection attempt.
```

plus mention and justification of the default value to provide a starting point for tuning.

Reply:
> - A quick skim of this wording seems to suggest that it should be set to a value around 0, which is incorrect.

I see how it could be interpreted as that, yeah.

> - Whether the number _of incoming connections_ hovers around zero or not is heavily dependent on the latency with which the application layer handles them, which we can't predict.

Ah, yeah, that's an error on my part.

> - Incoming connections might be consumed by means other than `accept`.

The method I linked was `Endpoint::accept`, not `Incoming::accept`, so it would still be on the path to refuse/retry/ignore. But it is a confusion hazard, yeah.

I largely rewrote the docs for these three settings, I think it should be more clear and correct now. Let me know what you think.